.CMake/alg_support.cmake

```diff
@@ -448,7 +448,7 @@ if(NOT ((OQS_MINIMAL_BUILD STREQUAL "") OR (OQS_MINIMAL_BUILD STREQUAL "OFF")))
 	filter_algs("${OQS_MINIMAL_BUILD}")
 elseif (${OQS_ALGS_ENABLED} STREQUAL "STD")
 ##### OQS_COPY_FROM_UPSTREAM_FRAGMENT_LIST_STANDARDIZED_ALGS_START
-	filter_algs("KEM_ml_kem_512_ipd;KEM_ml_kem_768_ipd;KEM_ml_kem_1024_ipd;SIG_dilithium_;SIG_dilithium_;SIG_dilithium_;SIG_ml_dsa_44_ipd;SIG_ml_dsa_65_ipd;SIG_ml_dsa_87_ipd;SIG_falcon_512;SIG_falcon_1024;SIG_sphincs_sha2_128f_simple;SIG_sphincs_sha2_128s_simple;SIG_sphincs_sha2_192f_simple;SIG_sphincs_sha2_192s_simple;SIG_sphincs_sha2_256f_simple;SIG_sphincs_sha2_256s_simple;SIG_sphincs_shake_128f_simple;SIG_sphincs_shake_128s_simple;SIG_sphincs_shake_192f_simple;SIG_sphincs_shake_192s_simple;SIG_sphincs_shake_256f_simple;SIG_sphincs_shake_256s_simple")
+	filter_algs("KEM_ml_kem_512_ipd;KEM_ml_kem_768_ipd;KEM_ml_kem_1024_ipd;SIG_ml_dsa_44_ipd;SIG_ml_dsa_65_ipd;SIG_ml_dsa_87_ipd;SIG_falcon_512;SIG_falcon_1024;SIG_sphincs_sha2_128f_simple;SIG_sphincs_sha2_128s_simple;SIG_sphincs_sha2_192f_simple;SIG_sphincs_sha2_192s_simple;SIG_sphincs_sha2_256f_simple;SIG_sphincs_sha2_256s_simple;SIG_sphincs_shake_128f_simple;SIG_sphincs_shake_128s_simple;SIG_sphincs_shake_192f_simple;SIG_sphincs_shake_192s_simple;SIG_sphincs_shake_256f_simple;SIG_sphincs_shake_256s_simple")
```

Comment:
(Still) not quite perfect: This apparently disables "KEM_ml_kem_512". Intentional? This is the reason for `oqsprovider` CI now failing. Positive is that this exercizes (kind-of successfully) `oqsprovider` without a single KEM active in `liboqs` (when built with STD config).

What about adding a test exercizing the "alias" feature (ascertaining that the same OQS_... defines get set)?  "oqsconfig.h" otherwise looks like this:

```
#define OQS_ENABLE_KEM_ml_kem_512_ipd 1
/* #undef OQS_ENABLE_KEM_ml_kem_512 */
...
``

Reply:
Thank you for catching this! I've now added the aliases to the fragment.
Ideally the alias would be automatically enabled in oqsconfig.h with a `cmake_dependent_option`. I'll look into this and a test next week.
But there seems another issue with `filter_algs` which appears to set everything to OFF except those explicitly provided.
For example, using the STD config I get
```
#define OQS_ENABLE_SIG_FALCON 1
#define OQS_ENABLE_SIG_falcon_512 1
/* #undef OQS_ENABLE_SIG_falcon_512_avx2 */
/* #undef OQS_ENABLE_SIG_falcon_512_aarch64 */
...
```
But I'd expect the aarch64-optimized version to be enabled on my laptop, which is set with the cmake default config.

Reply:
> I'll look into this and a test next week.
But there seems another issue with filter_algs which appears to set everything to OFF except those explicitly provided.

OK, #1703 created to track and unblock this PR.